Hide/Show register button FRONT

PedroR-dev opened the pull request:
Create logic to hide or show register button based on api flag. Update RegisterController to prevent sign up if button is hidden.
Limited permissions so that the admin can only modify this field, but not all domain config. Created new POE entries

PedroR-dev committed:
Create logic to hide or show register button based on api flag. Update RegisterController to prevent sign up if button is hidden.
Limited permissions so that the admin can only modify this field, but not all domain config. Created new POE entries

iduetxe commented on grails-app/controllers/springSecurity/RegisterController.groovy:
```diff
@@ -59,6 +57,8 @@ class RegisterController extends grails.plugin.springsecurity.ui.RegisterControl
         copy.remove 'action'
         def config = SpringSecurityUtils.securityConfig
 
+        registerButtonDisabledRedirect();
```
Rename to checkRegisterActiveAndRedirect

iduetxe commented on src/groovy/kuorum/web/admin/domain/DomainValidationCommand.groovy:
```diff
@@ -20,15 +20,16 @@ class DomainValidationCommand {
     static constraints = {
-        domainPrivacy nullable: false
-        firstFactorValidation nullable: false
+        domainPrivacy nullable: true
+        firstFactorValidation nullable: true
```
REvert

iduetxe commented on src/groovy/kuorum/web/admin/domain/DomainValidationCommand.groovy:
```diff
@@ -20,15 +20,16 @@ class DomainValidationCommand {
     Boolean isSocialNetwork
     Boolean isUserProfileExtended
     Boolean tourEnabled
+    Boolean showRegisterButton
```
Revert. As we talked this field will be in other command

PedroR-dev committed:
Reviewed requested changes. Change location of checkbox.
Remove previous logic and fields from DomainValidationCommand to DomainLandingCommand

PedroR-dev commented:
A futuro cuando se pueda me encantaría juntar la lógica de los dos checkbox, creo que no tiene sentido que estén separados

PedroR-dev committed:
Remove ROLE_ADMIN role annotation and unnecessary if statement
